src/infrastructure/adapters/PluginRegistry.ts

```
@@ -0,0 +1,517 @@
+            gracefulShutdownTimeout: 5000 as Milliseconds, // 5 seconds
+            ...config
+        }
+
+        Effect.log(`🔌 PluginRegistry initialized with config: ${JSON.stringify(this.config)}`)
+    }
+
+    /**
+     * Register and load a plugin
+     */
+    loadPlugin(plugin: Plugin): Effect.Effect<void, StandardizedError, never> {
+        const self = this
```

ESLint

[error] Unexpected aliasing of 'this' to local variable.

(@typescript-eslint/no-this-alias)